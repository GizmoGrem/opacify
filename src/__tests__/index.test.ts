import opacify, { hexToRGBA, RGBAToHexA, hexWithAlpha } from '../index';
import { ColorCode } from '../types';

test('HEX to HEXA', () => {
  expect(opacify.hexWithAlpha('#e61e4d', 0.5)).toBe('#e61e4d80');
});
test('HEX to HEXA with invalid params', () => {
  function invalidParams(hexColor: string, opacity: number) {
    hexWithAlpha(hexColor, opacity);
  }

  expect(() => invalidParams('#e61e4d', 12)).toThrowErrorMatchingSnapshot('opacity 12');
  expect(() => invalidParams('#e61e4jsnjkngd', 0.4)).toThrowErrorMatchingSnapshot('invalid color');
});

test('HEX to RGBA', () => {
  expect(opacify.hexToRgba('#e61e4d', 0.5)).toBe('rgba(230, 30 , 77, 0.5)');
});
test('HEX to RGBA with invalid params', () => {
  function invalidParams(hexColor: string, opacity: number) {
    hexToRGBA(hexColor, opacity);
  }

  expect(() => invalidParams('#e61e4d', 12)).toThrowErrorMatchingSnapshot('opacity 12');
  expect(() => invalidParams('#e61e4jsnjkngd', 0.4)).toThrowErrorMatchingSnapshot('invalid color');
});

test('RGB to HEXA', () => {
  expect(opacify.rgbaToHex(230, 30, 77, 0.5)).toBe('#e61e4d80');
});

test('RGB to HEXA with invalid params', () => {
  function invalidParams(r: ColorCode, g: ColorCode, b: ColorCode, opacity: number) {
    RGBAToHexA(r, g, b, opacity);
  }

  expect(() => invalidParams(230, 30, 77, 10)).toThrowErrorMatchingSnapshot('opacity 10');
});
