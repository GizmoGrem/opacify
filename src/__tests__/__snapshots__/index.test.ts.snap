// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HEX to HEXA with invalid params: invalid color 1`] = `"Invalid HEX param"`;

exports[`HEX to HEXA with invalid params: opacity 12 1`] = `"Invalid opacity, it should be between 0 and 1"`;

exports[`HEX to RGBA with invalid params: invalid color 1`] = `"Invalid HEX param"`;

exports[`HEX to RGBA with invalid params: opacity 12 1`] = `"Invalid opacity, it should be between 0 and 1"`;

exports[`RGB to HEXA with invalid params: opacity 10 1`] = `"Invalid opacity, it should be between 0 and 1"`;
